File: MyClub/MyClub.Model/Responses/PagedResult.cs
using System;
using System.Collections.Generic;

namespace MyClub.Model.Responses
{
    public class PagedResult<T>
    {
        public int TotalCount { get; set; }
        public int CurrentPage { get; set; }
        public int PageSize { get; set; }
        public int TotalPages => (int)Math.Ceiling((double)TotalCount / PageSize);
        public bool HasPrevious => CurrentPage > 0;
        public bool HasNext => CurrentPage < TotalPages - 1;
        public List<T> Data { get; set; } = new List<T>();
    }
}

File: MyClub/UI/myclub_desktop/lib/models/paged_result.dart
class PagedResult<T> {
  int totalCount;
  int currentPage;
  int pageSize;
  List<T> data;

  PagedResult({
    required this.totalCount,
    required this.currentPage,
    required this.pageSize,
    required this.data,
  });

  int get totalPages => (totalCount / pageSize).ceil();
  bool get hasPrevious => currentPage > 0;
  bool get hasNext => currentPage < totalPages - 1;

  factory PagedResult.fromJson(
    Map<String, dynamic> json,
    T Function(Map<String, dynamic>) fromJsonT,
  ) {
    return PagedResult<T>(
      totalCount: json['totalCount'] ?? 0,
      currentPage: json['currentPage'] ?? 0,
      pageSize: json['pageSize'] ?? 10,
      data: List<T>.from((json['data'] ?? []).map((x) => fromJsonT(x))),
    );
  }
}
